FieldOpt/Utilities/math.hpp

```diff
@@ -51,7 +51,9 @@ inline T calc_median(std::vector<T> list) {
     assert(!list.empty());
     std::sort(list.begin(), list.end());
     size_t size = list.size();
-    if (size % 2 == 0)
+    if (size == 1)
```

Comment:
It will be floored because they're integers. When you work with integers, you don't do rounding. You just ignore anything that would be after a decimal point (which is the same as flooring).